### PR TITLE
Impose a max length of 32 bytes on UTF-8 encoded aliases

### PR DESCRIPTION
**Pull request by araspitzu**
The configuration allowed the alias to be at most 32 chars but some chars are encoded in more than one byte (UTF-8) and this results in a spec violation where the alias was actually too long to be sent over the wire.

### CODE REVIEW

**Comment by araspitzu** [comment]
Fixes #819

File: eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala [hunk 1]
```diff
@@ -81,6 +81,9 @@ case class NodeParams(keyManager: KeyManager,
                       maxPendingPaymentRequests: Int,
                       maxPaymentFee: Double,
                       minFundingSatoshis: Long) {
+
+  require(alias.getBytes("UTF-8").length <= 32, "Invalid alias, too long")
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala]: They could be a  `require` here, but the actual message should be in `makeNodeParams` for consistency.

Also "alias must be shorter than 32B"  would be more informative

**Comment by araspitzu** [comment]
> The StartupSpec seems a bit over the top, you could instead load the default configuration, override the `node.alias` key and run `makeNodeParams`.

Addressed in 31854ba

File: eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala [hunk 2]
```diff
@@ -170,6 +171,9 @@ object NodeParams {
     val offeredCLTV = config.getInt("to-remote-delay-blocks")
     require(maxToLocalCLTV <= Channel.MAX_TO_SELF_DELAY && offeredCLTV <= Channel.MAX_TO_SELF_DELAY, s"CLTV delay values too high, max is ${Channel.MAX_TO_SELF_DELAY}")
 
+    val nodeAlias = config.getString("node-alias")
+    require(nodeAlias.getBytes("UTF-8").length <= 32, "Invalid alias, too long (max allowed 32 bytes)")
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/NodeParams.scala]:
```suggestion
    require(nodeAlias.getBytes("UTF-8").length <= 32, "invalid alias, too long (max allowed 32 bytes)")
```

**Pull request merged**